gcp/iap-ingress/base/config-map.yaml

```diff
@@ -1,349 +1,74 @@
-                                                "timeout_ms": 10000,
-                                                "use_websocket": true,
-                                                "weighted_clusters": {
-                                                    "clusters": [
-                                                        {
-                                                            "name": "cluster_ambassador",
-                                                            "weight": 100
-                                                        }
-                                                    ]
-                                                }
-                                            }
-                                        ]
-                                    }
-                                ]
-                            },
-                            "stat_prefix": "ingress_http"
-                        },
-                        "name": "http_connection_manager",
-                        "type": "read"
-                    }
-                ]
-            }
-        ],
-        "stats_flush_interval_ms": 1000,
-        "statsd_udp_ip_address": "127.0.0.1:8025"
-    }
+  healthcheck_route.yaml: |
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: default-routes
+      namespace: $(namespace)
+    spec:
+      hosts:
+      - "*"
+      gateways:
+      - kubeflow-gateway
+      http:
+      - match:
+        - uri:
+            exact: /healthz
+        route:
+        - destination:
+            port:
+              number: 80
+            host: whoami-app.kubeflow.svc.cluster.local
+      - match:
+        - uri:
+            exact: /whoami
+        route:
+        - destination:
+            port:
+              number: 80
+            host: whoami-app.kubeflow.svc.cluster.local
+    ---
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: kubeflow-gateway
+      namespace: $(namespace)
+    spec:
+      selector:
+        istio: ingressgateway
+      servers:
+      - port:
+          number: 80
+          name: http
+          protocol: HTTP
+        hosts:
+        - "*"
+  jwt-policy-template.yaml: |
+    apiVersion: authentication.istio.io/v1alpha1
+    kind: Policy
+    metadata:
+      name: ingress-jwt
+      namespace: $(istioNamespace)
+    spec:
+      targets:
+      - name: istio-ingressgateway
+        ports:
+        - number: 80
+      origins:
+      - jwt:
+          issuer: https://cloud.google.com/iap
+          jwksUri: https://www.gstatic.com/iap/verify/public_key-jwk
+          audiences:
+          - {{JWT_AUDIENCE}}
+          jwtHeaders:
+          - x-goog-iap-jwt-assertion
+          trigger_rules:
+          - excluded_paths:
+            - exact: /healthz
+            - prefix: /.well-known/acme-challenge
+      principalBinding: USE_ORIGIN
   setup_backend.sh: |
```

Comment:
original file for this is here: https://github.com/kubeflow/kubeflow/blob/master/kubeflow/gcp/setup_backend.sh

I'd like to use patch for this file.  in the meantime, if you wanna update the logic, could you also update the original file?

also, kubeflow/kubeflow#3298 changes the script, could you also update them here?

Reply:
not sure I followed your suggestions. 

The [original file](https://raw.githubusercontent.com/kubeflow/kubeflow/master/kubeflow/gcp/setup_backend.sh) is using [patch](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/gcp/setup_backend.sh#L58)

Reply:
sorry for the confusion - I meant patch of the whole `setup_backend.sh`.  Currently we hardcoded these scripts for config map, right?  Previously we maintain these scripts in separate files and import them when deployment.  I guess the analogy would be patching these scripts in kustomize?  But in any case this is not something we should include into this PR.